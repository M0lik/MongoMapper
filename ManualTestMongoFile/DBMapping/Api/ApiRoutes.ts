import * as express from "express";

import GroupUserRoutes from './GroupUser';
import UserRoutes from './User';
import DishesRoutes from './Dishes';
import IngredientsRoutes from './Ingredients';

let router = express.Router();

router.use('/GroupUser', GroupUserRoutes);
router.use('/User', UserRoutes);
router.use('/Dishes', DishesRoutes);
router.use('/Ingredients', IngredientsRoutes);

export default router;