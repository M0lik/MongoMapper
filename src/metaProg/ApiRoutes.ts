import * as fs from 'fs';

/**
 *
 *
 * @export
 * @class FileWriter
 */
export default class ApiRoutes {
    private _file: string;
    private _collectionsNames: string[];
    /**
     *Creates an instance of FileWriter.
     * @param {string} fileName the name of the mongoose schema, it will write 'name + Schema'
     * @param {string} schema the schema structure to write
     * @memberof FileWriter
     */
    constructor(fileNames: string[]) {
        this._collectionsNames = fileNames;
        this._file = this.header() + this.routes() + this.footer();
    }

    private header = () => {
        let retStr: string = `import * as express from "express";\n\n`;

        for (const name of this._collectionsNames) {
            retStr += `import ${name}Routes from './${name}';\n`;
        }
        
        return retStr + `\nlet router = express.Router();\n\n`
    }

    private footer = () => `\nexport default router;`;

    private routes = () => {
        let result: string = '';

        for (const name of this._collectionsNames) {
            result += `router.use('/${name}', ${name}Routes);\n`;
        }

        return result;
    };

    getFileAsString = () => this._file;
    write = (pathToWriteFile: string) => fs.writeFileSync(pathToWriteFile, this._file, { encoding: 'utf8', flag: 'w' });
}