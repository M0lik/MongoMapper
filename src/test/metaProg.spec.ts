import ApiRoutes from "../metaProg/ApiRoutes";
import ApiWriter from "../metaProg/apiWriter";
import FileWriter from "../metaProg/fileWriter";

describe('Api Routes', () => {
    const fileName: string[] = ['Test1', 'Test2']; 

    test('Test filename', () => {
        let l = new ApiRoutes(fileName);
        let a = l.getFileAsString();
        expect(a).toMatchSnapshot();
    })
});

describe('Api Writer', () => {
    const fileName: string = 'Test1'; 

    test('Test filename', () => {
        let l = new ApiWriter(fileName);
        let a = l.getFileAsString();
        expect(a).toMatchSnapshot();
    })
});

describe('file writer', () => {
    const fileName: string = 'Test1'; 
    const schema: any = {title : 'titre'};

    test('Test whithout filename', () => {
        let l = new FileWriter(fileName, schema);
        let a = l.getFileAsString();
        expect(a).toMatchSnapshot();
    })
});