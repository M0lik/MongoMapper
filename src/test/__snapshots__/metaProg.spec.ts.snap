// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Api Routes Test filename 1`] = `
"import * as express from \\"express\\";

import Test1Routes from './Test1';
import Test2Routes from './Test2';

let router = express.Router();

router.use('/Test1', Test1Routes);
router.use('/Test2', Test2Routes);

export default router;"
`;

exports[`Api Writer Test filename 1`] = `
"import Test1 from '../Schema/Test1';

import * as express from 'express';

let router = express.Router();

router.get('/', async (req, res) => {
        let result = await Test1.readByField();
        res.send(result);
    });

router.get('/:id', async (req, res) => {
        let result = await Test1.readById(req.params.id);
        res.send(result);
    });

router.post('/', async (req, res) => {
        let create = await Test1.create(req.body);
        res.sendStatus(200);
    });

router.delete('/:id', async (req, res) => {
        let result = await Test1.delete(req.params.id);
        res.send(result);
    });

export default router;"
`;

exports[`file writer Test whithout filename 1`] = `
"var mongoose = require('mongoose');
var Schema = mongoose.Schema;

var Test1Schema = new Schema({title:titre});

const Test1Data = mongoose.model(\\"Test1\\", Test1Schema, \\"Test1\\");

const Test1 = {
create : async (objToCreate : any) => (new Test1Data(objToCreate)).save(),

readByField : async (objFilter : any = {}) => await Test1Data.find(objFilter),
readById : async (dataId : string) => mongoose.Types.ObjectId.isValid(dataId) ? (await Test1Data.find({_id : dataId}))[0] : null,

update : async (document : any) => await document.save(),

delete : async (dataId : any) => mongoose.Types.ObjectId.isValid(dataId) ? await Test1Data.deleteOne({_id : dataId}) : null,

}

export default Test1;"
`;
